src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -167,7 +167,29 @@ function BaseSelectionList<TItem extends ListItem>(
     const {isKeyboardShown} = useKeyboardState();
     const [itemsToHighlight, setItemsToHighlight] = useState<Set<string> | null>(null);
     const itemFocusTimeoutRef = useRef<NodeJS.Timeout | null>(null);
-    const [currentPage, setCurrentPage] = useState(1);
+    const isItemSelected = useCallback(
+        (item: TItem) => item.isSelected ?? ((isSelected?.(item) ?? selectedItems.includes(item.keyForList ?? '')) && canSelectMultiple),
+        [isSelected, selectedItems, canSelectMultiple],
+    );
+    const calculateInitialCurrentPage = useCallback(() => {
+        if (canSelectMultiple || sections.length === 0) {
+            return 1;
+        }
+
+        let currentIndex = 0;
+        for (const section of sections) {
+            if (section.data) {
+                for (const item of section.data) {
+                    if (isItemSelected(item)) {
+                        return Math.floor(currentIndex / CONST.MAX_SELECTION_LIST_PAGE_LENGTH) + 1;
+                    }
+                    currentIndex++;
+                }
+            }
+        }
+        return 1;
+    }, [canSelectMultiple, isItemSelected, sections]);
```

Comment:
Can you please add plain english comment for why we doing this so its easier for people to follow?

Reply:
Sure, added one at the top of the function